bokeh/document.py

```diff
@@ -359,6 +365,16 @@ def roots(self):
         return list(self._roots)
 
     @property
+    def session_context(self):
+        return self._session_context
+
+    @session_context.setter
+    def session_context(self, session_context):
+        if self._session_context != session_context:
+            self._session_context = session_context
+            self._trigger_on_change(SessionContextChangedEvent(self, session_context))
```

Comment:
I agree it makes no senses for it to change. I'd be ok with: 

* only a getter and no setter
* the code that "attaches" the session context setting the private attr directly (with a comment)